webui/react/src/pages/TrialDetails/TrialInfoBox.tsx

```diff
@@ -50,6 +53,33 @@ const TrialInfoBox: React.FC<Props> = ({ trial, experiment }: Props) => {
         return `${trial?.logRetentionDays} days`;
     }
   }, [trial]);
+  const integrationData = useMemo(() => {
+    const {
+      config: { integrations },
+    } = experiment;
+
+    if (integrations?.pachyderm !== undefined) {
+      const url = createPachydermLineageLink(integrations.pachyderm);
+
+      return {
+        hasIntegrationData: url !== undefined ? true : false,
+        text: '<MLDM repo>',
+        url,
+      };
+    }
+
+    return;
+  }, [experiment]);
+
+  const handleClickDataInput = useCallback(
+    (e: AnyMouseEvent) => {
+      if (integrationData?.hasIntegrationData)
+        handlePath(e, {
+          path: integrationData.url,
+        });
+    },
+    [integrationData],
+  );
```

Comment:
consider memoizing the entire component or only memoizing the url.

* in the integrationData, `hasIntegrationData` and `text` will always be the same value in the object -- we don't need to memoize them.
* moving the memoization up to include the entire `OverviewStats` component or down to only the url means that we no longer need to check the integration data again on user click.